### PR TITLE
remove unused function negotiate() and writeYAML()

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
remove unused function negotiate() and writeYAML() in k8s.io/apiserver/pkg
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go [hunk 1]
```diff
@@ -119,32 +119,6 @@ func isPrettyPrint(req *http.Request) bool {
 	return false
 }
 
-// negotiate the most appropriate content type given the accept header and a list of
-// alternatives.
-func negotiate(header string, alternatives []string) (goautoneg.Accept, bool) {
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go]: @sttts a while back, we had negotiation problems.  Is this intentionally removed?

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go]: Looks suspicious. @hanxiaoshuai can you find out when we stopped using this func?

Comment (hanxiaoshuai) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go]: func negotiate() was used in func negotiateOutput in v1.4.x , but  func negotiateOutput was remove in v1.5.x. So func negotiate is not used any more. @deads2k @sttts

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go]: Ok, that's long ago.

**Pull request merged**